src/serlio/prtModifier/PRTModifierAction.cpp

```diff
@@ -132,19 +132,18 @@ std::list<MObject> getNodeAttributesCorrespondingToCGA(const MFnDependencyNode&
 		// maya annoyance: color attributes automatically get per-component plugs/child attrs
 		if (attr.isUsedAsColor()) {
 			MFnCompoundAttribute compAttr(attrObj);
-			for (size_t ci = 0, numChildren = compAttr.numChildren(); ci < numChildren; ci++)
+			for (unsigned int ci = 0, numChildren = compAttr.numChildren(); ci < numChildren; ci++)
 				ignoreList.emplace_back(compAttr.child(ci));
 		}
 
 		rawAttrs.push_back(attrObj);
 	}
 
-	std::list<MObject> filteredAttrs;
-	for (const auto& a: rawAttrs) {
-		if (std::find(ignoreList.begin(), ignoreList.end(), a) == ignoreList.end())
-			filteredAttrs.push_back(a);
-	}
-	return filteredAttrs;
+	std::remove_if(rawAttrs.begin(), rawAttrs.end(), [&ignoreList] (const auto& attr) {

```

Comment:
i think this is the classic trap, std::remove_if does not remove anything without a followup std::erase, see https://en.cppreference.com/w/cpp/algorithm/remove

Reply:
Ouch, that's embarrassing!

Reply:
C++20 to the rescue: https://en.cppreference.com/w/cpp/container/vector/erase2

Reply:
Yeah, I found this as well when I was looking for the erase-remove idiom. But I realize just now, this is for `std::vector`, for `std::list` there's already [`std::list::remove_if`](https://en.cppreference.com/w/cpp/container/list/remove) that actually **erases** the matching elements. That's what confused me in the first place. Then again, I think there's no reason to use a `std::list` here instead of a `std::vector` 🤔

Reply:
there is also a free function for std::list: https://en.cppreference.com/w/cpp/container/list/erase2